scripts/preprocessor.py

```diff
@@ -6,212 +6,240 @@
+
+    cellsX = len(attributeToValueMap[intToAttribute[0]])
+    cellsY = len(attributeToValueMap[intToAttribute[1]])
+    output = categorical_2d(
+        dataset,
+        list(attribute_ids.keys()),
+        attribute_type_map,
+        attributeToValueMap,
+        attribute_ids,
+        mesh_codes_to_ids)
+
+  with open(output_directory + '/' + computation_request_id + '.txt', 'w') as f:
+      for item in output:
+          f.write("%s\n" % item)
+
+  with open(output_directory + '/' + computation_request_id + '.txt', "rb") as f:
+      SHA256 = sha256()
+      for byte_block in iter(lambda: f.read(4096), b""):
+          SHA256.update(byte_block)
+
+  json_output = {'precision': '{0:.{1}f}'.format(10**(-decimal_accuracy), decimal_accuracy),
+                  'sizeAlloc': sizeAlloc,
+                  'cellsX': cellsX,
+                  'cellsY': cellsY,
+                  'dataSize': dataset_size,
+                  'hash256': SHA256.hexdigest(),
+                  'attributeToInt': attributeToInt,
+                  'intToAttribute': intToAttribute}
+
+  with open(output_directory + '/' + computation_request_id + '.json', 'w') as f:
+      json.dump(json_output, f, indent=4)
 
 
 def main():
-    parser = argparse.ArgumentParser(
-        description='SMPC local drive preprocessor')
-    parser.add_argument(
-        '-c',
-        '--computation',
-        required=True,
-        type=str,
-        help='The ID of the computation (required)')
-    parser.add_argument(
-        '-d',
-        '--dataset',
-        required=True,
-        type=str,
-        help='Dataset file (required)')
-    parser.add_argument(
-        '-m',
-        '--mapping',
-        required=True,
-        type=str,
-        help='Mapping file (required)')
-    parser.add_argument(
-        '-a',
-        '--attributes',
-        nargs='*',
-        help='List of space seperated attributes which will be included in the output file (default: all)')
-    parser.add_argument(
-        '-g',
-        '--algorithm',
-        default='1d_categorical_histogram',
-        choices=[
-            '2d_mixed_histogram',
-            '1d_categorical_histogram',
-            '1d_numerical_histogram',
-            'secure_aggregation',
-            '2d_numerical_histogram',
-            '2d_categorical_histogram'],
-        type=str,
-        help='Computation algorithm (default: %(default)s)')
-    parser.add_argument(
-        '-p',
-        '--precision',
-        default=5,
-        type=int,
-        help='The number of decimal digits to be consider for floats (default: %(default)s)')
-    parser.add_argument('--version', action='version', version='%(prog)s 0.1')
-    args = parser.parse_args()
-    preprocess(args.algorithm, args.computation, args.attributes, args.dataset, args.mapping, args.precision)  # , filters = {"Medical Record Number":filter1, "RVEDV (ml)": filter2})
+  parser = argparse.ArgumentParser(
```

Comment:
If we decide to put the files (  mapping_file_name, mesh_codes_to_ids_file, mesh_ids_to_codes_file) as arguments then they must declared as command line arguments also